core/src/main/java/org/elasticsearch/monitor/fs/FsProbe.java

```diff
@@ -136,7 +136,11 @@ public FsInfo stats(FsInfo previous, @Nullable ClusterInfo clusterInfo) throws I
     }
 
     /* See: https://bugs.openjdk.java.net/browse/JDK-8162520 */
-    private static long adjustForHugeFilesystems(long bytes) {
+    /**
+     * Take a large value intended to be positive, and if it has overflowed,
+     * return {@code Long.MAX_VALUE} instead of a negative number.
+     */
+    public static long adjustForHugeFilesystems(long bytes) {
```

Comment:
Sure, I'll make it package-private, I just figured it's static, so there's no harm in it being public

Updated in 52ee78f191e1515570ae9e16d611920f336562c0